zcash_keys/src/keys.rs

```diff
@@ -807,18 +884,20 @@ impl UnifiedFullViewingKey {
                         .map_err(|_| AddressGenerationError::InvalidTransparentChildIndex(_j))?,
                 );
             } else {
-                return Err(AddressGenerationError::KeyNotAvailable(Typecode::P2pkh));
+                return Err(AddressGenerationError::KeyNotAvailable(Typecode::P2PKH));
             }
         }
         #[cfg(not(feature = "transparent-inputs"))]
         let transparent = None;
 
-        UnifiedAddress::from_receivers(
+        UnifiedAddress::new(
             #[cfg(feature = "orchard")]
             orchard,
             #[cfg(feature = "sapling")]
             sapling,
             transparent,
+            std::cmp::min(self.expiry_height, request.expiry_height),
+            std::cmp::min(self.expiry_time, request.expiry_time),
         )
         .ok_or(AddressGenerationError::ShieldedReceiverRequired)
```

Comment:
We should make sure that an invalid combination isn't representable as a `UnifiedAddressRequest`, so that this error is impossible (and can be a panic if it does occur due to some inconsistency).

Reply:
This is certainly the wrong error to return at this point, but what the correct error is depends upon https://github.com/zcash/librustzcash/pull/1135#discussion_r1515444301

Reply:
Resolved in 673e1a21d28209103dce50ede358491e5978d007